src/rules/pydocstyle/rules/imperative_mood.rs

```diff
@@ -0,0 +1,37 @@
+use imperative::Mood;
+use once_cell::sync::Lazy;
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::docstrings::definition::Docstring;
+use crate::registry::Diagnostic;
+use crate::rules::pydocstyle::helpers::{logical_line, normalize_word};
+use crate::violations;
+
+static MOOD: Lazy<Mood> = Lazy::new(Mood::new);
+
+/// D401
+pub fn imperative_mood(checker: &mut Checker, docstring: &Docstring) {
+    let body = docstring.body;
+
+    if let Some(index) = logical_line(body) {
```

Comment:
Unfortunately I think this is the index of the _last_ physical line in the logical line. So, like, if the "first logical line" extends beyond a single physical line, like:

```
"""Here's a logical line that
extends to two physical lines.
"""
```

Then the index would be 1 instead of 0.

We can either modify `logical_line` to return the (start, end) indices. _Or_, you could just do `body.trim().split_whitespace().next()`? You just want the very first word after all.

Reply:
(Otherwise, LGTM.)

Reply:
Changed to trim the docstring, then get the first line, and the first word off there – the error message looks better when it has the context of the full first line 👍